src/manifold/src/edge_op.cpp

```diff
@@ -576,14 +587,14 @@ void Manifold::Impl::RecursiveEdgeSwap(const int edge,
     const glm::vec2 e23 = v[3] - v[2];
     if (glm::dot(e23, e23) < precision_ * precision_) {
       tag++;
-      CollapseEdge(tri0edge[2]);
+      CollapseEdge(tri0edge[2], edges);
+      edges.resize(0);
     } else {
       visited[edge] = tag;
       visited[pair] = tag;
-      RecursiveEdgeSwap(tri0edge[0], visited, tag);
-      RecursiveEdgeSwap(tri0edge[1], visited, tag);
-      RecursiveEdgeSwap(tri1edge[0], visited, tag);
-      RecursiveEdgeSwap(tri1edge[1], visited, tag);
+      // note: the order should be inverted
```

Comment:
This is a touch ambiguous - do you mean it is currently correct, or should be changed?

Reply:
it is currently correct